TextToSpeech/impl/NetworkStatusObserver.cpp

```diff
@@ -0,0 +1,113 @@
+#include <WPEFramework/securityagent/SecurityTokenUtil.h>
+#endif
+
+#define MAX_SECURITY_TOKEN_SIZE 1024
+#define NETWORK_CALLSIGN "org.rdk.Network"
+#define NETWORK_CALLSIGN_VER NETWORK_CALLSIGN".1"
+
+namespace WPEFramework {
+namespace Plugin {
+
+NetworkStatusObserver* NetworkStatusObserver::getNetworkStatusObserver()
+{
+    static NetworkStatusObserver *instance = new NetworkStatusObserver();
+    return instance;
+}
+
+string NetworkStatusObserver::getSecurityToken()
+{
+    std::string token = "token=";
+    int tokenLength = 0;
+    unsigned char buffer[MAX_SECURITY_TOKEN_SIZE] = {0};
+    static std::string endpoint;
+
+    if(endpoint.empty()) {
+       Core::SystemInfo::GetEnvironment(_T("THUNDER_ACCESS"), endpoint);
+       TTSLOG_INFO("Thunder RPC Endpoint read from env - %s", endpoint.c_str());
+    }
+
+    if(endpoint.empty()) {
+        Core::File file("/etc/WPEFramework/config.json", false);
+        if(file.Open(true)) {
+            JsonObject config;
+            if(config.IElement::FromFile(file)) {
+                Core::JSON::String port = config.Get("port");
+                Core::JSON::String binding = config.Get("binding");
+                if(!binding.Value().empty() && !port.Value().empty())
+                    endpoint = binding.Value() + ":" + port.Value();
+            }
+        file.Close();
+        }
+
+        if(endpoint.empty()) {
+            endpoint = _T("127.0.0.1:9998");
+            TTSLOG_INFO("Thunder RPC Endpoint read from config file - %s", endpoint.c_str());
+            Core::SystemInfo::SetEnvironment(_T("THUNDER_ACCESS"), endpoint);
+        }
+     }
+
+     string payload = "http://localhost";
+     if(payload.empty()) {
+          tokenLength = GetSecurityToken(sizeof(buffer), buffer);
+     } else {
+          int buffLength = std::min(sizeof(buffer), payload.length());
+          ::memcpy(buffer, payload.c_str(), buffLength);
+          tokenLength = GetToken(sizeof(buffer), buffLength, buffer);
+     }
+
+     if(tokenLength > 0) {
+         token.append((char*)buffer);
+     } else {
+         token.clear();
+     }
+
+     TTSLOG_INFO("Thunder token - %s", token.empty() ? "" : token.c_str());
+     return token;
+}
+
+
+bool NetworkStatusObserver::isConnected()
+{
+    if (m_networkService == nullptr) {
+        string token = getSecurityToken();
+        m_networkService = new WPEFramework::JSONRPC::LinkType<Core::JSON::IElement>(_T(NETWORK_CALLSIGN_VER),"", false, token);
+
+        JsonObject joGetParams, joGetResult;
+        uint32_t status = m_networkService->Invoke<JsonObject, JsonObject>(1000, "isConnectedToInternet", joGetParams, joGetResult);
+        if (status > 0) {
+            TTSLOG_ERROR("%s call failed %d",NETWORK_CALLSIGN_VER, status);
+            return false;
+        } else if (joGetResult.HasLabel("connectedToInternet")) {
+             m_isConnected=joGetResult["connectedToInternet"].Boolean();
+        }
```

Comment:
If "connectedToInternet" label is not present, should we be resetting the m_isConnected to false?